lib/Controller/ConfigController.php

```diff
@@ -197,10 +164,128 @@ public function setAdminConfig(array $values): DataResponse {
 				$this->clearUserInfo($user->getUID());
 			});
 		}
+	}
 
+	/**
+	 * returns auto-generated OAuth client information for nextcloud
+	 *
+	 * @return array<mixed>
+	 */
+	private function createAndGetClientInformationForNextCloud(): array {
+		$this->deleteOauthClient();
+		$opUrl = $this->config->getAppValue(Application::APP_ID, 'oauth_instance_url', '');
+		$clientInfo = $this->oauthService->createNcOauthClient('OpenProject client', rtrim($opUrl, '/') .'/oauth_clients/%s/callback');
+		$this->config->setAppValue(Application::APP_ID, 'nc_oauth_client_id', $clientInfo['id']);
+		return $clientInfo;
+	}
+
+	/**
+	 * set config values
+	 * @NoAdminRequired
+	 *
+	 * @param array<string, string> $values
+	 * @return DataResponse
+	 */
+	public function setConfig(array $values): DataResponse {
+		foreach ($values as $key => $value) {
+			$this->config->setUserValue($this->userId, Application::APP_ID, $key, trim($value));
+		}
+		$result = [];
+
+		if (isset($values['token'])) {
+			if ($values['token'] && $values['token'] !== '') {
+				$result = $this->storeUserInfo();
+			} else {
+				$this->clearUserInfo();
+				$result = [
+					'user_name' => '',
+				];
+			}
+		}
+		if (isset($result['error'])) {
+			return new DataResponse($result, Http::STATUS_UNAUTHORIZED);
+		} else {
+			return new DataResponse($result);
+		}
+	}
+
+	/**
+	 *
+	 * set admin config values
+	 * @param array<string, string|null> $values
+	 *
+	 * @return DataResponse
+	 */
+	public function setAdminConfig(array $values): DataResponse {
+		try {
+			$this->setUpOrResetUpOpenProjectAdminConfig($values);
+		} catch (\InvalidArgumentException $e) {
+			return new DataResponse([
+				'error' => $this->l->t($e->getMessage())
+			], HTTP::STATUS_BAD_REQUEST);
+		}
 		return new DataResponse([
 			"status" => OpenProjectAPIService::isAdminConfigOk($this->config)
-		]);
+		], HTTP::STATUS_OK);
+	}
+
+	/**
+	 * set up whole integration
+	 *
+	 * @NoCSRFRequired
+	 * @param array<string, string|null> $values
+	 *
+	 * @return DataResponse
+	 */
+	public function setUpWholeIntegration(array $values): DataResponse {
+		// response array to send for this request
+		$responseToSend = [];
+		$mustHaveClientKey = [
+			'oauth_instance_url',
+			'client_id',
+			'client_secret',
+		];
+
+		foreach ($mustHaveClientKey as $keyName) {
+			if (!array_key_exists($keyName, $values)) {
+				$responseToSend['op_setup_info'] = [
+					"status" => "Missing Open Project client info or Invalid Key"
+				];
+				return new DataResponse($responseToSend, HTTP::STATUS_BAD_REQUEST);
+			}
+			if ($keyName == "default_enable_navigation" || $keyName == "default_enable_unified_search") {
+				$values[$keyName] = (bool)$values[$keyName];
+			}
+			$values[$keyName] = (string)$values[$keyName];
```

Comment:
```suggestion
			if ($keyName == "default_enable_navigation" || $keyName == "default_enable_unified_search") {
				$values[$keyName] = (bool)$values[$keyName];
			} else {
			        $values[$keyName] = (string)$values[$keyName];
		        }
```
otherwise it will be cast to bool and then to string